### PR TITLE
Feature - 4857 - Self declaration form

### LINKED ISSUE #4857: Build Self Declaration form (no carousel)
## Description
Build the Honestly Pledge form (without carousel) in storybook.

### Screenshot 
Screen Shot 2022-11-21 at 11:25 AM)

### Figma Prototype
https://www.figma.com/proto/IThJdlxAJqQPJyvX88hAG7/IAP-Honesty-Pledge-V6?node-id=4521%3A20259&scaling=min-zoom&page-id=4521%3A20212&starting-point-node-id=4521%3A20259

### Acceptance Criteria 

- [ ] 1.  Initially only shows the Self-Declaration radio buttons as required field with below values 

       -   “I affirm that I am First Nations (Status or Non-Status), Inuk (Inuit), and or a Métis person”
       -  “I am not a member of an Indigenous group, and I would like to see other opportunities available to me”
- [ ] 2.  When clicking on "I affirm ..." the Select you Community(ies) component and the boxed My Communities section underneath should be displayed.
- [ ] 3. When clicking on "I am not ..."  it should be displayed like below and link sends you to [GC Digital Talent Homepage](https://talent.canada.ca/en/) 
Screen Shot 2022-11-17 at 10 26 01 AM
- [ ] 4. When any of the community checkboxes are selected, the icon becomes coloured, and the name of is added in a Pill in the My Communities box.
- [ ] 5. If "I am First Nations" is selected, the radio selection appears below for "I am Status First Nations" and "I am Non-Status First Nations" goes from being empty and disabled to being required. If "I am First Nations" is uncheck, the radio selection goes back to disabled and empty.
- [ ] 6. When "I am Indigenous and I don't see my community here" is selected, its pill is coloured yellow and an informational alert appears: 
Informational alert
- [ ] 7. When the informational alert is closed, the pill should no longer be coloured yellow.
After closing alert
- [ ] 8. When "I affirm" is selected, and at least one community is selected, a Signature box and submit button (labeled "Sign and continue") should appear under the form.
- [ ] 9. Accessibility review from Rosalie or Maddy

### PR DESCRIPTION
**Pull request by esizer**
## 👋 Introduction

This adds the Indigenous Self-Declaration form (in storybook) so it can eventually be added to a page and connected to the API.

## 📋 TO DO

- [x] Get designer review/sign off
- [x] Write tests for cases noted
- [x] Send for translations
- [x] Add translations

## ✏️ Meeting Notes

-  [x] Alert stays shown even after deselecting "I don't see my community here"

## 🧪 Testing

1. Run storybook `npm run storybook --workspace=indigenousapprenticeship `
2. Navigate to the "Self-Declaration Form" story
3. Ensure the following scenarios:
     - Only the "Self Declaration" radio group appears on initial load
     - Disclaimer linking to browse page appears if not identified as Indigenous
     - Icon checkboxes appear after identifying as Indigenous
     - Icons light up when selecting a community
     - Chips appear when selecting a community
     - Dismissing chips removes check on community
     - If "Other" and any other option is selected together and alert appears and the "Other" chip is yellow
     - Dismissing alert changes chip to primary colour
     - First Nations status field appears if user identifies as First Nations
     - Signature and submit button only appear if identified as Indigenous and at least one community is selected

## 🤖 Robot Stuff

Resolves #4857

### CODE REVIEW

**Review by yonikid15 — CHANGES_REQUESTED** [review]
Awesome work Eric, your code is beautiful stuff 👍 
I'm still in the middle of the review, but I found a small problem with the SignAndContinue section of the form, which off the bat I'm not sure really sure of a simple fix. The submit button falls out of line, when an error is triggered. 

image

**Comment by esizer** [comment]
> I'm still in the middle of the review, but I found a small problem with the SignAndContinue section of the form, which off the bat I'm not sure really sure of a simple fix. The submit button falls out of line, when an error is triggered.

Yeah, I noticed that as well. I'll make a note of this and when I connect with @BrookeCameron we can discuss some solutions to this. I now the submit button was an issue that surfaced during the a11y testing (unrelated to this though)

**Review by yonikid15 — CHANGES_REQUESTED** [review]
I'm seeing the CommunitySelection component is not responsive.
image

Should be an easy fix by reducing the padding/margin in mobile view.

However, everything else is looking really great Eric 👍. Like we discussed before, I agree that having a form with no submit button is weird. I'll review this again once those possible issues are addressed.

File: frontend/indigenousapprenticeship/src/js/components/SelfDeclaration/utils.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      }),
+    ],
+    [
+      "firstNations",
+      intl.formatMessage({
+        id: "GSqmjE",
+        defaultMessage: "First Nations",
+        description: "Label for First Nations community",
+      }),
+    ],
```
[the remark this answers is not in the document]
Comment (esizer) [reply frontend/indigenousapprenticeship/src/js/components/SelfDeclaration/utils.ts]: Removed in [remove duplicate key in community map](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4869/commits/25635eeb8571735a8aa25f70849140dd1c7f02be)

[the remark this answers is not in the document]
Comment (esizer) [reply frontend/indigenousapprenticeship/src/js/components/SelfDeclaration/utils.ts]: > I'm seeing the CommunitySelection component is not responsive.

@yonikid15 The form should now be good to look at on smaller screen sizes. Only think left is to finalize the copy and send off for translations 😃

**Comment by esizer** [comment]
> I'm seeing the CommunitySelection component is not responsive.

@yonikid15 The form should now be good to look at on smaller screen sizes. Only think left is to finalize the copy and send off for translations 😃

**Review by yonikid15 — APPROVED** [review]
Looking great eric! Last thing are those translations and its good to go 👍

**Review by mnigh — CHANGES_REQUESTED** [review]
My review is mostly focused on grammar questions and suggestions but there are a few copy related questions as well that might need to asked to the author of the original English copy. 

⚠️ A warning on some of my suggestions, especially any changes to English `intl` strings: this will change the id and the `fr.json` equivalent id would also need to be changed.

File: apps/web/src/components/SelfDeclaration/SignAndContinue.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+              })}
+            />
+          </div>
+        </>
+      ) : (
+        <>
+          <p data-h2-margin="base(x1, 0)">
+            {intl.formatMessage({
+              defaultMessage:
+                "This Program is for First Nations, Inuit, and Métis peoples within the geographic boundaries of Canada.",
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/components/SelfDeclaration/SignAndContinue.tsx]: Updated in [grammar updates](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4869/commits/e7978738f9a515c5ad48598d0c3fdf25a1bf13a7)

File: apps/web/src/components/SelfDeclaration/CommunitySelection.tsx [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+                'Are you sure you meant to select "I am Indigenous and I don\'t see my community here?"',
+              id: "at11n5",
+              description:
+                "Title for the alert warning users about selection not represented and a represented community",
+            })}
+          </Alert.Title>
+          <p>
+            {intl.formatMessage({
+              defaultMessage:
+                "The Program is for First Nations, Inuit, and Métis peoples within the geographic boundaries of Canada. At a later step of the application process, you may be asked to provide proof that you are Indigenous. You can read more about how this will be verified below.",
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/components/SelfDeclaration/CommunitySelection.tsx]: Updated in [grammar updates](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4869/commits/e7978738f9a515c5ad48598d0c3fdf25a1bf13a7)

File: apps/web/src/components/SelfDeclaration/SelfDeclarationForm.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+              description: "Title for the indigenous self-declaration form",
+            })}
+          </Heading>
+          <p
+            data-h2-text-align="base(center)"
+            data-h2-margin="base(x1, 0, x2, 0)"
+          >
+            {intl.formatMessage({
+              defaultMessage:
+                "We recognize the importance of Indigenous voices in the federal government. The Program was designed in partnership with Indigenous peoples. By completing and signing the Indigenous Peoples Self-Declaration Form, you are helping to protect the space, agreeing that you are a part of the three distinct Indigenous groups in Canada, and are interested in joining the Program!",
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/components/SelfDeclaration/SelfDeclarationForm.tsx]: Updated in [grammar updates](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4869/commits/e7978738f9a515c5ad48598d0c3fdf25a1bf13a7)

File: apps/web/src/components/SelfDeclaration/SelfDeclarationForm.tsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+                id: "7STO48",
+                description:
+                  "Text for the option to self-declare as Indigenous",
+              }),
+            },
+            {
+              value: "no",
+              label: intl.formatMessage({
+                defaultMessage:
+                  '"I am not a member of an Indigenous group, and I would like to see other opportunities available to me"',
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/components/SelfDeclaration/SelfDeclarationForm.tsx]: Updated in [grammar updates](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4869/commits/e7978738f9a515c5ad48598d0c3fdf25a1bf13a7)

File: apps/web/src/lang/fr.json [hunk 6]
```diff
@@ -363,6 +363,170 @@
...
+    "qEvJjr": {
+      "defaultMessage": "Afin de garantir que le programme reste accessible aux peuples autochtones du Canada, vous devez remplir le formulaire d’auto-déclaration des peuples autochtones et confirmer que vous êtes membre d’une ou plusieurs des communautés autochtones du Canada",
+      "description": "Paragraph three for the self-declaration explanation dialog"
+    },
+    "qwJkEy": {
+      "defaultMessage": "Ce programme s’adresse aux membres de Premières nations, aux Inuits et aux Métis du Canada.",
+      "description": "Disclaimer displayed when a user has indicated they are not Indigenous on the self-declaration form."
+    },
+    "r6L5aI": {
+      "defaultMessage": "« Auto-déclaration »",
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/lang/fr.json]: Updated in [grammar updates](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4869/commits/e7978738f9a515c5ad48598d0c3fdf25a1bf13a7)

File: apps/web/src/lang/fr.json [hunk 7]
```diff
@@ -363,6 +363,170 @@
...
+    "GSqmjE": {
+      "defaultMessage": "Premières Nations",
+      "description": "Label for First Nations community"
+    },
+    "KaisAC": {
+      "defaultMessage": "Métis",
+      "description": "Label for Métis community"
+    },
+    "Md1Pj4": {
+      "defaultMessage": "Autochtone selon la définition de ce programme :",
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/lang/fr.json]: Updated in [grammar updates](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4869/commits/e7978738f9a515c5ad48598d0c3fdf25a1bf13a7)

File: apps/web/src/pages/IAPHomePage/Dialog/DefinitionDialog.tsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    <Dialog.Root>
+      <Dialog.Trigger>
+        <Button color="ia-secondary" mode="inline" {...btnProps}>
+          {children}
+        </Button>
+      </Dialog.Trigger>
+      <Dialog.Content>
+        <Dialog.Header color="ia-primary">
+          {intl.formatMessage({
+            defaultMessage: "Indigenous as defined for this program:",
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/IAPHomePage/Dialog/DefinitionDialog.tsx]: Updated in [grammar updates](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4869/commits/e7978738f9a515c5ad48598d0c3fdf25a1bf13a7)

File: apps/web/src/lang/fr.json [hunk 9]
```diff
@@ -419,7 +419,7 @@
       "defaultMessage": "Métis",
       "description": "Trigger text for Métis tab"
     },
-    "Cn9LYM": {
+    "Ap86I/": {
       "defaultMessage": " Ce programme s’adresse aux membres des Premières Nations, aux Inuits et aux Métis qui se trouvent au Canada. Lors d’une étape ultérieure de la procédure de demande, vous aurez peut-être à fournir une preuve que vous êtes une personne autochtone. Vous pouvez en savoir plus sur la vérification de votre statut ci-dessous.",
```
Comment (mnigh) [inline apps/web/src/lang/fr.json:423]: Remove extra space
```suggestion
      "defaultMessage": "Ce programme s’adresse aux membres des Premières Nations, aux Inuits et aux Métis qui se trouvent au Canada. Lors d’une étape ultérieure de la procédure de demande, vous aurez peut-être à fournir une preuve que vous êtes une personne autochtone. Vous pouvez en savoir plus sur la vérification de votre statut ci-dessous.",
```

**Review by mnigh — CHANGES_REQUESTED** [review]
Might be out of scope, but I noticed the top left and right corners of the dialogs are not rounded though the [designs](https://www.figma.com/proto/IThJdlxAJqQPJyvX88hAG7/IAP-Honesty-Pledge?node-id=4521%3A20259&scaling=min-zoom&page-id=4521%3A20212&starting-point-node-id=4521%3A20259) are rounded on all corners:

## Screenshot
Screen Shot 2023-01-24 at 09 48 51

File: apps/web/src/components/SelfDeclaration/SelfDeclarationForm.tsx [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+        <Separator
+          orientation="horizontal"
+          decorative
+          data-h2-background-color="base(ia-secondary)"
+          data-h2-margin="base(x2, 0)"
+        />
+        <div data-h2-text-align="base(center)">
+          <HelpLink />
+          <p data-h2-font-weight="base(700)" data-h2-margin="base(x1, 0)">
+            {intl.formatMessage(
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/components/SelfDeclaration/SelfDeclarationForm.tsx:128]: Updated in [update dialog link colour](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4869/commits/dc44104bad137f9929a2641d05e51de9b60944fe)

File: apps/web/src/pages/IAPHomePage/Dialog/VerificationDialog.tsx [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+          {intl.formatMessage({
+            defaultMessage: "How will this be verified?",
+            id: "h4E9+K",
+            description: "Heading for the self-declaration verification dialog",
+          })}
+        </Dialog.Header>
+        <p data-h2-margin="base(x1, 0)">
+          {intl.formatMessage({
+            defaultMessage:
+              "At a later step of the application process, you may be asked to provide proof that you are Indigenous. Some of the ways to do this were included in the definition statement at the beginning of this application, including a Certificate of Indian Status card, an Inuit beneficiary card, a Metis citizenship card, a letter from an authorized representative of a recognized Indigenous community, in addition to an attestation form. For the purpose of inclusion, these measures are to ensure that this program remains available to Indigenous Peoples in Canada.",
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/src/pages/IAPHomePage/Dialog/VerificationDialog.tsx]: Updated this in [fix Métis typo](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4869/commits/d86fa9919a4c6c2ac8466b5c1a0a3cd86ed9ca00) to make sure the ID was updated

**Pull request merged**